include/ignition/physics/detail/SpecifyData.hh

```diff
@@ -39,6 +39,23 @@ namespace ignition
       // Do nothing
     }
 
+    /////////////////////////////////////////////////
+    template <typename Expected>
+    ExpectData<Expected>::ExpectData(const ExpectData<Expected> &_other)
+        : ExpectData()
+    {
+      this->Copy(_other);
+    }
+
+    /////////////////////////////////////////////////
+    template <typename Expected>
+    ExpectData<Expected> &ExpectData<Expected>::operator=(
+        const ExpectData<Expected> &_other)
+    {
+      this->Copy(_other);
```

Comment:
Similar to the previous comment, I think doing this will re-copy all the data of the `CompositeData` many times.

As far as I can figure, an explicit copy assignment operator should not be needed. If the compiler doesn't produce one automatically (I can't remember the rules about that), then we should be able to simply do

```
ExpectedData &operator=(const ExpectedData &_other) = default;
```

in the class definition of `ExpectedData`.

Reply:
Agreed. I had to add an empty copy assignment operator to `PrivateSpecifyData` for this to work because the compiler doesn't generate one due to `PrivateSpecifyData::expectedIterator` being a const. https://github.com/ignitionrobotics/ign-physics/pull/77/commits/ec3947b318120951011ec1982c1c81f8abd63ffd